examples/threading/threading.c

```diff
@@ -14,20 +14,49 @@ void* threadfunc(void* thread_param)
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
     //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
-    return thread_param;
+    
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    
+    DEBUG_LOG("Waiting till lock\n");
+    usleep((thread_func_args->obtain_ms)*1000);
+    pthread_mutex_lock(thread_func_args->mutex);
+
+    DEBUG_LOG("holding till release\n");
+    usleep((thread_func_args->release_ms)*1000);
+    pthread_mutex_unlock(thread_func_args->mutex);
+    
+    DEBUG_LOG("thread complete\n");
+    thread_func_args->thread_complete_success=true;
```

Comment:
@kamini2801 did the thread complete successfully if each of the steps above failed?